src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -141,91 +141,103 @@ public TextWriter Log
 
         private bool _isFallbackBodyCompilation; // True if we're compiling a fallback method body after compiling the real body failed
 
+#if READYTORUN
+        private bool ShouldSkipCompilation(IMethodNode methodCodeNodeNeedingCode)
+        {
+            return methodCodeNodeNeedingCode.Method.IsAggressiveOptimization;
+        }
+#endif
+
         private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, MethodIL methodIL = null)
         {
+            bool shouldSkipCompilation = false;
 #if READYTORUN
             bool codeGotPublished = false;
+            shouldSkipCompilation = ShouldSkipCompilation(methodCodeNodeNeedingCode);
 #endif
             try
             {
-                _isFallbackBodyCompilation = methodIL != null;
+                if (!shouldSkipCompilation)
```

Comment:
The extra bool in ifdefs and big if block doesn't help readability. Why not just put it in `CompileMethod` (the place that calls this method)?

```csharp
        public void CompileMethod(IReadyToRunMethodCodeNode methodCodeNodeNeedingCode)
        {
            _methodCodeNode = methodCodeNodeNeedingCode;

            if (!ShouldSkipCompilation(methodCodeNodeNeedingCode))
                CompileMethodInternal(methodCodeNodeNeedingCode);
            else
                PublishEmptyCode();
        }
```

Also move `ShouldSkipCompilation` method to CorInfoImpl.ReadyToRun.cs.